cartridge

```diff
@@ -2030,6 +2053,43 @@ local function check_pid_running(pid)
     return ffi.C.kill(tonumber(pid), 0) == 0
 end
 
+local function read_configuration(path)
+    if fio.path.is_dir(path) then
+        local configs = fun.chain(
+            fio.glob(fio.pathjoin(path, '*.yml')),
+            fio.glob(fio.pathjoin(path, '*.yaml'))
+        ):map(function(x) return read_configuration(x) end):totable()
+        return fun.chain(unpack(configs)):tomap()
+    else
+        return yaml.decode(read_file(path))
+    end
+end
+
+-- Read configuration at `path` and fetch instance names.
+local function get_configured_isntances(path, app_name)
+    assert(app_name and #app_name > 0, 'app_name must be given')
```

Comment:
Oh, I see, It actually asserts that app_name must be given. But I think this should be done on the `parse` step.

Reply:
I don't think so. There it tries to figure out app name when it was not explicitly given. It just returns nothing when it can not do this.

Reply:
But you don't check if there is `app_name` given, when instance_name provided. I.e:

```
cartridge start .instance_name
```

Reply:
It was not required at that time. I've changed behaviour, app_name is always required and it's checked in `.parse` method.